pkg/opnictl/commands/create_cluster.go

```diff
@@ -1,26 +1,381 @@
 package commands
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"net/url"
+	"os"
+	"strings"
 
+	"github.com/AlecAivazis/survey/v2"
+	nfdv1 "github.com/kubernetes-sigs/node-feature-discovery-operator/api/v1"
+	"github.com/mattn/go-isatty"
+	"github.com/rancher/opni/apis/v1beta1"
+	"github.com/rancher/opni/pkg/features"
+	"github.com/rancher/opni/pkg/opnictl/common"
+	"github.com/rancher/opni/pkg/providers"
+	cliutil "github.com/rancher/opni/pkg/util/opnictl"
 	"github.com/spf13/cobra"
 	"github.com/ttacon/chalk"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type flagVars struct {
+	edit                    bool
+	name                    string
+	version                 string
+	provider                string
+	interactive             bool
+	noPretrainedModel       bool
+	noLogAdapter            bool
+	noNodeFeatureDiscovery  bool
+	noGpuPolicyAdapter      bool
+	controlPlaneModelSource string
+	controlPlaneModelImage  string
+	controlPlaneModelURL    string
+}
+
+var (
+	defaultModelURL = "https://opni-public.s3.us-east-2.amazonaws.com/pretrain-models/control-plane-model-v0.1.2.zip"
+)
+
+func buildPretrainedModel(cmd *cobra.Command, vars *flagVars) (*v1beta1.PretrainedModel, error) {
+	pretrainedModel := &v1beta1.PretrainedModel{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "control-plane",
+			Namespace: common.NamespaceFlagValue,
+		},
+	}
+	switch vars.controlPlaneModelSource {
+	case "http":
+		if vars.controlPlaneModelURL == "" {
+			// interactive check is done in PreRunE
+			err := survey.AskOne(
+				&survey.Input{
+					Message: "Enter a URL for the control-plane pretrained model",
+					Default: defaultModelURL,
+				},
+				&vars.controlPlaneModelURL,
+				survey.WithValidator(survey.ComposeValidators(
+					survey.Required,
+					func(ans interface{}) error {
+						url, err := url.Parse(ans.(string))
+						if err != nil {
+							return err
+						}
+						if url.Scheme != "http" && url.Scheme != "https" {
+							return fmt.Errorf("URL scheme must be http or https")
+						}
+						return nil
+					},
+				)),
+			)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Should we just hardcode this seeing as we are responsible for producing and publishing the control plane model?

Reply:
This would be for airgap support